neqo-transport/src/connection/tests/datagram.rs

```diff
@@ -224,6 +225,76 @@ fn datagram_acked() {
     ));
 }
 
+fn send_packet_and_check_server_event(
+    client: &mut Connection,
+    server: &mut Connection,
+    event: &ConnectionEvent,
+) {
+    let out = client.process_output(now()).dgram();
+    server.process_input(out.unwrap(), now());
+    let events: Vec<_> = server
+        .events()
+        .filter_map(|evt| match evt {
+            ConnectionEvent::RecvStreamReadable { .. } | ConnectionEvent::Datagram { .. } => {
+                Some(evt)
+            }
+            _ => None,
+        })
+        .collect();
+    // We should only get one event - either RecvStreamReadable or Datagram.
+    assert_eq!(events.len(), 1);
+    assert_eq!(mem::discriminant(event), mem::discriminant(&events[0]));
```

Comment:
If you change this function to return the event, you can call it like this:

```rust
    assert!(matches!(send_packet_and_get_server_event(&mut client, &mut server), ConnectionEvent::RecvStreamReadable { stream_id: s } if s == stream_id);
```

The problem with only matching the discriminant is that you aren't testing that the stream ID is correct.

Reply:
Thanks. Will do.